addons/purchase/models/purchase.py

```diff
@@ -1010,28 +1010,26 @@ def _get_date_planned(self, seller, po=False):
     @api.depends('product_id', 'date_order')
     def _compute_account_analytic_id(self):
         for rec in self:
-            if not rec.account_analytic_id:
-                default_analytic_account = rec.env['account.analytic.default'].sudo().account_get(
-                    product_id=rec.product_id.id,
-                    partner_id=rec.order_id.partner_id.id,
-                    user_id=rec.env.uid,
-                    date=rec.date_order,
-                    company_id=rec.company_id.id,
-                )
-                rec.account_analytic_id = default_analytic_account.analytic_id
+            default_analytic_account = rec.env['account.analytic.default'].sudo().account_get(
```

Comment:
You should only compute it if there is no `display_type`, like done in `sale`.
And you should apply this fix to `sale` too

Reply:
You means as it is done ine here:
https://github.com/odoo/odoo/blob/c7dd2815760ca1f7e3bc0abae8d9f0bcd6b57420/addons/sale/models/sale.py#L1662-L1663
?

Reply:
Like here for instance (in master, but it should be the same on all versions)
https://github.com/odoo/odoo/blob/c5291f4cf459224a9ebbecb397f7efdd415b2448/addons/sale/models/sale_order_line.py#L884